carta/image.py

```diff
@@ -426,23 +426,103 @@ def set_polarization(self, polarization, recursive=True):
+    def set_center(self, x, y, system=None):
+        """Set the center position, in image or world coordinates. Optionally change the session-wide coordinate system.
 
-        TODO: what are the units?
+        Coordinates must either both be image coordinates or match the current number formats. Numbers and numeric strings with no units are interpreted as degrees.
 
         Parameters
         ----------
         x : {0}
             The X position.
         y : {1}
             The Y position.
+        system : {2}
+            The coordinate system. If this parameter is provided, the coordinate system will be changed session-wide before the X and Y coordinates are parsed.
+
+        Raises
+        ------
+        ValueError
+            If image coordinates are outside the bounds of the image, if a mix of image and world coordinates is provided, or if world coordinates do not match the session-wide number format.
+        """
+        if system is not None:
+            self.session.set_coordinate_system(system)
+
+        x_is_pixel = PixelValue.valid(str(x))
+        y_is_pixel = PixelValue.valid(str(y))
+
+        if x_is_pixel and y_is_pixel:
+            # Image coordinates
+            x_value = PixelValue.as_float(str(x))
+            y_value = PixelValue.as_float(str(y))
+            if 0 <= x_value < self.width and 0 <= y_value < self.height:
+                self.call_action("setCenter", x_value, y_value)
+            else:
+                raise ValueError(f"Image coordinates ({x_value}, {y_value}) are outside the bounds of the image ({self.width} x {self.height}).")
+
+        elif x_is_pixel or y_is_pixel:
+            raise ValueError("Cannot mix image and world coordinates.")
+
+        else:
+            if not self.valid_wcs:
+                raise ValueError("Cannot parse world coordinates. This image does not contain valid WCS information.")
+
+            number_format_x, number_format_y, _ = self.session.number_format()
+            x_value = WorldCoordinate.with_format(number_format_x).from_string(str(x), SpatialAxis.X)
+            y_value = WorldCoordinate.with_format(number_format_y).from_string(str(y), SpatialAxis.Y)
+            self.call_action("setCenterWcs", str(x_value), str(y_value))
+
+    @validate(Size())
+    def zoom_to_size_x(self, size):
+        """Zoom to the given X size.
+
+        Numbers and numeric strings with no units are interpreted as arcseconds.
+
+        Parameters
+        ----------
+        size : {0}
+        """
+        size = str(size)
+        if PixelValue.valid(size):
+            self.call_action("zoomToSizeX", PixelValue.as_float(size))
+        else:
+            if not self.valid_wcs:
+                raise ValueError("Cannot parse angular size. This image does not contain valid WCS information.")
+            self.call_action("zoomToSizeXWcs", str(AngularSize.from_string(size)))
+
+    @validate(Size())
+    def zoom_to_size_y(self, size):
```

Comment:
Wonder if we can combine zoom_to_size_x and zoom_to_size_y as zoom_to_size and have an argument as "direction" (or "dimension" or "axis"?) for x or y? This would simplify the code repeatition here.

Reply:
Good idea. I already added a `SpatialAxis` enum for the `set_center` range validation, so this is a simple change.